packages/wallet/backend/src/webMonetization/transaction/service.ts

```diff
@@ -53,24 +53,28 @@ export class WMTransactionService implements IWMTransactionService {
     })
   }
 
-  async deleteByTransactionIds(ids: string[]) {
-    return WMTransaction.query().del().whereIn('id', ids)
+  async deleteByTransactionIds(ids: string[], trx?: TransactionOrKnex) {
+    return WMTransaction.query(trx).del().whereIn('id', ids)
   }
 
-  async sumByPaymentPointerId(paymentPointerId: string, type: TransactionType) {
-    const transactions = await WMTransaction.query().where({
+  async sumByPaymentPointerId(
+    paymentPointerId: string,
+    type: TransactionType,
+    trx?: TransactionOrKnex
+  ) {
+    const transactions = await WMTransaction.query(trx).where({
       paymentPointerId,
       type,
       status: 'COMPLETED'
     })
     const ids = transactions.map(({ id }) => id)
-    const sumResult = (await WMTransaction.query()
+    const sumResult = (await WMTransaction.query(trx)
       .whereIn('id', ids)
       .sum('value')) as unknown as [{ sum: bigint }]
 
     return {
       ids,
-      sum: sumResult[0].sum
+      sum: sumResult[0].sum ?? 0n
```

Comment:
I know we don't enjoy doing calculations in JS, but since everything is bigint format, my take is that we'd end up fine with summing everything in JS and avoid another query.
But I'm fine with it this way as well